src/enveloping/CommitmentValidator.ts

```diff
@@ -0,0 +1,41 @@
+import log from 'loglevel'
+import { Commitment, CommitmentReceipt } from './Commitment'
+import { ethers } from 'ethers'
+
+/**
+ * The CommitmentValidator can verify a CommitmentReceipt signature
+ */
+export class CommitmentValidator {
+  /**
+   * Validates a CommitmentReceipt signature
+   * @param receipt the CommitmentReceipt to be validated
+   * @return true or false
+   */
+  validateCommitmentSig (receipt: CommitmentReceipt | undefined): boolean {
```

Comment:
sorry, but it is unclear to me the cases in which a receipt would come with `undefined` type